File: python/src/context.py
from provider import Provider

class Context:
    def __init__(self, client, node, id):
        self.client = client
        self.node = node
        self.id = id

    def provider(self, profile):
        args = [self.node.system.id, self.node.id, self.id, profile]
        transaction = self.client.send('json', 'providers', args)
        self.client.wait_for_response(transaction)
        return Provider(self.client, self, profile)

File: nodejs/context.js
// Copyright 2025 Padi, Inc. All Rights Reserved.

export class Context {
  #client;
  #node;
  #id;

  constructor(client, node, id) {
    this.#client = client;
    this.#node = node;
    this.#id = id;
  }
}
